providers/implementations/kdfs/hkdf.c

```diff
@@ -117,17 +134,33 @@ static void kdf_hkdf_free(void *vctx)
     KDF_HKDF *ctx = (KDF_HKDF *)vctx;
 
     if (ctx != NULL) {
-        kdf_hkdf_reset(ctx);
+        kdf_hkdf_reset_ex(vctx, 1);
         OPENSSL_free(ctx);
     }
 }
 
 static void kdf_hkdf_reset(void *vctx)
+{
+    kdf_hkdf_reset_ex(vctx, 0);
+}
+
+static void kdf_hkdf_reset_ex(void *vctx, int full_reset)
 {
     KDF_HKDF *ctx = (KDF_HKDF *)vctx;
     void *provctx = ctx->provctx;
+    int preserve_digest = full_reset ? 0 : ctx->fixed_digest;
+    PROV_DIGEST digest = { 0 };
 
-    ossl_prov_digest_reset(&ctx->digest);
+    /*
+     * TODO: OPENSSL MAINTAINERS: should this instead use ossl_prov_digest_copy() to back up the
```

Comment:
I think what you have done is the best you can do..
maybe change the name to save_prov_digest...

Reply:
Rather than embed this note in the PR you can also comment on your PR...

Reply:
Just change the comment to something like
/* For  fixed digests just save and restore the PROV_DIGEST object */